urllib3/connection.py

```diff
@@ -299,12 +302,26 @@ def connect(self):
                     'for details.)'.format(hostname)),
                     SubjectAltNameWarning
                 )
-            match_hostname(cert, self.assert_hostname or hostname)
+            _match_hostname(cert, self.assert_hostname or hostname)
 
         self.is_verified = (resolved_cert_reqs == ssl.CERT_REQUIRED or
                             self.assert_fingerprint is not None)
 
 
+def _match_hostname(cert, asserted_hostname):
```

Comment:
Also this function is very simple right now, but I was anticipating that it could become a bit more complex, if @Lukasa or someone else added the ability to verify certificates for IP addresses by backporting all of the stuff from Python 3.5, as @Lukasa mentioned in https://github.com/shazow/urllib3/pull/820#issuecomment-197956174